test/mpsc_queue/stm_mpsc_queue.ml

```diff
@@ -0,0 +1,147 @@
+(** Sequential and Parallel model-based tests of mpsc_queue *)
+
+open QCheck
+open STM
+open Util
+module Mpsc_queue = Lockfree.Mpsc_queue
+
+module WSDConf = struct
+  type cmd = Push of int | Pop | Push_head of int | Is_empty | Close
+
+  let show_cmd c =
+    match c with
+    | Push i -> "Push " ^ string_of_int i
+    | Pop -> "Pop"
+    | Push_head i -> "Push_head" ^ string_of_int i
+    | Is_empty -> "Is_empty"
+    | Close -> "Close"
+
+  type state = bool * int list
+  type sut = int Mpsc_queue.t
+
+  let producer_cmd _s =
+    let int_gen = Gen.nat in
+    QCheck.make ~print:show_cmd
+      (Gen.oneof
+         [
+           Gen.map (fun i -> Push i) int_gen;
+           Gen.return Is_empty;
+           Gen.return Close;
+         ])
+
+  let arb_cmd _s =
+    let int_gen = Gen.nat in
+    QCheck.make ~print:show_cmd
+      (Gen.oneof
+         [
+           Gen.return Pop;
+           Gen.map (fun i -> Push_head i) int_gen;
+           Gen.return Is_empty;
+           Gen.return Close;
+         ])
+
+  let init_state = (false, [])
+  let init_sut () = Mpsc_queue.create ()
+  let cleanup _ = ()
+
+  let next_state c (is_closed, s) =
+    match c with
+    | Push i ->
+        (is_closed, if not is_closed then i :: List.rev s |> List.rev else s)
+    | Push_head i -> (is_closed, if not (is_closed && s = []) then i :: s else s)
+    | Is_empty -> (is_closed, s)
+    | Pop -> ( (is_closed, match s with [] -> s | _ :: s' -> s'))
```

Comment:
double parens here

Reply:
This is actually done by `ocamlformat`. Pretty weird. I raised an issue in ocamlfomat repo.